### PR TITLE
feat: summary workflow for multiple plots

### LINKED ISSUE #221: Summary workflow: How to handle more than 2 metrics per benchmarking event
Currently the metrics calculation and subsequent consolidation steps expect one "metric_x" and one "metric_y" value per benchmarking event, which are then plotted against each other. However, in APAeval we are planning to calculate more than 2 metrics in one summary workflow run, and consequently also to create more than one plot.    

Possible solutions:
- adapt `compute_metrics.py`, `manage_assessment_data.py` and possibly other scripts to create `.json` output files with more than 2 metrics. This will most likely break OEB compatibility. New scripts for plotting will thus also have to be created.
- duplicate and adapt the summary workflows, such that for each challenge (dataset) multiple SWFs have to be run to compute multiple metrics/create multiple plots (e.g. SWF1 computes metrics 1 & 2 and plots them against each other; SWF2 computes metric 3 and creates a barplot participants vs. metric 3; SWF3 computes metrics 4 & 5,...). This will not break OEB compatibility, but the number of challenges will be heavily inflated (in the above example number of challenges will be 3 times more than before)

### PR DESCRIPTION
**Pull request by ninsch3000**
In APAeval we want to plot multiple plots per challenge, which was not possible with the initial summary workflow. Hence several scripts of the workflow have to be adapted. Some scripts will be replaced as maintenance became infeasible. Specifically, `manage_assessment_data.py` will be replaced by `aggregation.py`. A template for aggregation files will be introduced.

TO DO
- [x] complete `aggregation.py` (To dos as comments inside script)
- [x] set up aggregation template plus handling
- [x] adapt `assessment_charts.py`
- [x] adapt `merge_data_model_files.py`
- [x] integrate with `main.nf` and rest of summary wf
- [ ] write more To Dos

Fixes #221 #226

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated corresponding READMEs (if applicable)
- [x] My code follows the templates/style guidelines of the repository
- [x] In- and output formats comply with APAeval specifications
- [x] No parameters or file names are hardcoded
- [x] Results, logs or other output is not commited to the repository
- [x] I have tested my code

### CODE REVIEW

**Review by dominikburri — APPROVED** [review]
Hi @ninsch3000, many thanks :) 

I only have minor comments, resp. questions.

I tested it locally with the supplied `Qx.json` and an own `assessment_datasets.json`, which worked as expected: it added the new tool into `Qx.json`.
Also tested when there was no previous aggregation file (Qx.json), and instead needed to use the template `aggregation_template_Q.json`, this also worked as intended. 

Then moved on to test the whole summary workflow. For this I built the local docker images as described in the readme. 
First ran the example configured in `quantification/nextflow.config`, which worked without problems.
Then I used results from PAQR (results don't matter, just wanted a reasonable BED file), adjusted the config accordingly with challenge id AK and ran nextflow. I got a reasonable `consolidated_result.json` and `assessment_datasets.json`. The individual results and plots also looked good. 
Then I copied `PAQR_benchmark_results/results/AK/AK.json` into `APAeval_Q_test_data/data` as the benchmarking data (*assess_dir* in config), renamed the tool, changed the metric values and ran again the same nextflow workflow. Now, as intended, not the template was used but this `AK.json` and the resulting aggregation file `PAQR_benchmark_results/results/AK/AK.json` had both tools insides. The generated plots also displayed both tools. 

To conclude, based on my testing (primarely `aggregation.py`) the summary workflow for multiple plots runs as intended.

File: summary_workflows/quantification/quantification_dockers/q_consolidation/aggregation.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        with open(aggregation_file, mode='w', encoding="utf-8") as f:
+            json.dump(new_aggregation, f, sort_keys=True, indent=4, separators=(',', ': '))
+        
+        # 2.f) Write assessments per challenge to local results dir
+        # We have stored the assessment json objects for each challenge in the challenges dict
+        challenge_assessments = []
+        for metric, ass_json in challenges[challenge_id].items():
+            challenge_assessments.append(ass_json)
+        
+        assessment_file = os.path.join(challenge_dir, participant_id + ".json")
```
[the remark this answers is not in the document]
Comment (ninsch3000) [reply summary_workflows/quantification/quantification_dockers/q_consolidation/aggregation.py:158]: Yes, the same as "assessment_datasets.json", it's duplicated for the viz in the VRE to work (or some other OEB VRE related reason)

**Review by dominikburri — APPROVED** [review]
Thank you @ninsch3000 and @AsierGonzalez for the work!
As the new changes are minor, I quickly went over. I think it is good to be merged from my side.

**Review by dominikburri — APPROVED** [review]
Hi @ninsch3000 and @AsierGonzalez, thank you for the hard work!

I tested the workflow with the provided test data and it ran successfully.
I also ran some real life samples and checked with @ninsch3000 the results. First, participant1 was ran on two challenges based off the provided template, then the aggregation objects from `consolidated_result.json` was moved to new files to be recognised by *assess_dir* for the second participant. The resulting plots and aggregation objects contained both participants. All seemed reasonable. 

What we noticed and need to be aware of: each metric and window size combination is computed and plotted. This should not be a problem, but we might not need each plot on OEB.

So from my side with the successful execution of local runs, it is fine to be merged.
I would still like to have some feedback from (Meritxell or @AsierGonzalez) how it runs on the VRE.

**Review by AsierGonzalez — APPROVED** [review]

**Comment by AsierGonzalez** [comment]
Clarification: In the VRE the online mode for the old benchmarking does NOT work because there is data for the challenge in the db but it has different field names. I don't think this should block the merge though, it's fixed if we move to the new benchmarking event, which is the one we should be using anyway

**Pull request merged**